src/v/cluster/topics_frontend.cc

```diff
@@ -334,6 +334,25 @@ ss::future<topic_result> topics_frontend::do_update_topic_properties(
           std::move(update.tp_ns), errc::resource_is_being_migrated};
     }
 
+    auto id_validation_enabled = [&update]() -> bool {
+        const auto& props = update.properties;
+        return std::ranges::any_of(
+          std::to_array({
+            props.record_key_schema_id_validation,
+            props.record_key_schema_id_validation_compat,
+            props.record_value_schema_id_validation,
+            props.record_value_schema_id_validation_compat,
+          }),
+          [](const auto& p) {
+              return p.op == incremental_update_operation::set
+                     && p.value.value_or(false);
```

Comment:
With AlterConfig, unrelated changes will be declined. We should allow unrelated changes.

Reply:
See [this commit](https://github.com/redpanda-data/redpanda/pull/24058/commits/1d325ab04a2e5a82bc26110123ad693c08f3145e#diff-8e018e56b4ed06ab759458400f1a7629c2a18bddca81bec76e0c371a5133b748) for some refactoring that allows to see if a config has become set.

Reply:
Yeah. I'm leaning twd dropping this bit entirely, at least until your stuff lands. Better to concentrate discussion in one place.